core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```diff
@@ -100,11 +100,14 @@ object ConsoleConsumer extends Logging {
       val msg: BaseConsumerRecord = try {
         consumer.receive()
       } catch {
-        case e: Throwable => {
+        case nse: NoSuchElementException =>
```

Comment:
This may be too general and can hide other problems. Could we throw an EndOfStreamException in OldConsumer.receive() and catch that?